terraform/shared/modules/grafana/cognito.tf

```diff
@@ -0,0 +1,42 @@
+resource "aws_cognito_user_pool" "pool" {
+  name = "grafana"
+
+  mfa_configuration = "ON"
+
+  password_policy {
+    minimum_length                   = 12
+    require_lowercase                = true
+    require_uppercase                = true
+    require_numbers                  = true
+    require_symbols                  = true
+    temporary_password_validity_days = 7
+  }
+
+  software_token_mfa_configuration {
+    enabled = true
+  }
+
+  verification_message_template {
+    default_email_option = "CONFIRM_WITH_LINK"
+    email_subject        = "Verify access to Grafana"
+    email_message        = "Click the link below to verify access to Grafana \n {####}"
```

Comment:
I assume this is some mysterious magic incantation?

Reply:
Ooogily Boogily now you're in cognito